compileopts/options.go

```diff
@@ -26,6 +27,7 @@ type Options struct {
 	GOMIPS          string // environment variable (only used with GOARCH=mips and GOARCH=mipsle)
 	Directory       string // working dir, leave it unset to use the current working dir
 	Target          string
```

Comment:
Worth adding `BuildMode` to `TargetSpec` as well, so `wasm-unknown` and `wasip2` can set build mode to `c-shared` by default?

Reply:
We can do that in the PR that adds `//go:wasmexport` to `wasm-unknown` :)

Reply:
Ok. I'll make a follow up PR for wasip2 and unknown.